docs/design/design_v1.10.md

```diff
@@ -0,0 +1,523 @@
+     - `kube-controller-manager.yaml`
+     - `kube-scheduler.yaml`
+ - `/etc/kubernetes/kubelet.conf` as the path where the kubelet should store its credentials to the API server.
+ - `/etc/kubernetes/admin.conf` as the path from where the admin can fetch his/her superuser credentials.
+ - Names of certificates files:
+   - `ca.crt`, `ca.key` (CA certificate)
+   - `apiserver.crt`, `apiserver.key` (API server certificate)
+   - `apiserver-kubelet-client.crt`, `apiserver-kubelet-client.key` (client certificate for the apiservers to connect to the kubelets securely)
+   - `sa.pub`, `sa.key` (a private key for signing ServiceAccount )
+   - `front-proxy-ca.crt`, `front-proxy-ca.key` (CA for the front proxy)
+   - `front-proxy-client.crt`, `front-proxy-client.key` (client cert for the front proxy client)
+- Names of kubeconfig files:
+  - `admin.conf`
+  - `kubelet.conf` (`bootstrap-kubelet.conf` during TLS bootstrap)
+  - `controller-manager.conf`
+  - `scheduler.conf`
+
+
+## `kubeadm init` phases
+
+`kubeadm init` internal workflow consists of a sequence of atomic work tasks to perform.
+
+The `kubeadm alpha phase` command allows users to invoke individually each task, and ultimately offers a reusable and composable API/toolbox that can be used by other Kubernetes bootstrap tools / by any IT automation tool / by advanced user for creating custom clusters. 
+
+### Preflight checks
+
+`kubeadm` executes a set of preflight checks before starting the init, with the aim to verify preconditions and avoid common cluster startup problems. 
+In any case the user can skip specific preflight checks (or eventually all preflight checks) with the `--ignore-preflight-errors` option.
+
+- [warning] If the Kubernetes version to use (passed with the `--kubernetes-version` flag) is at least one minor version higher than the kubeadm CLI version.
+- Kubernetes system requirements:
+  - if running on linux:
+    - [error] if not Kernel 3.10+ or 4+ with specific KernelSpec.
+    - [error] if required cgroups subsystem aren't in set up.
+  - if using docker:
+    - [error/warning] if Docker endpoint does not exists or does not work, if docker version >17.03.
+    Note: starting from 1.9, kubeadm provides better support for CRI-generic functionality; in that case, docker specific controls are skipped or replaced by similar controls for crictl
+- [error] if user is not root.
+- [error] if hostname is not a valid DNS subdomain; [warning] if the host name cannot be reached via network lookup.
+- [error] if kubelet version is lower that the minimum kubelet version supported by kubeadm (current minor -1).
+- [error] if kubelet version is at least one minor higher than the required controlplane version (unsupported version skew).
+- [warning] if kubelet service does not exists, if it is disabled.
+- [warning] if firewalld is active.
+- [error] if API.BindPort or ports 10250/10251/10252 are used.
+- [Error] if /etc/kubernetes/manifest folder already exists and it is not empty.
```

Comment:
This is not quite right now since https://github.com/kubernetes/kubernetes/pull/57287 was just merged. Also, a typo: `manifest` --> `manifests`

Reply:
@mattkelly That's expected. This is just an exact copy of the v1.9 design, and I'm now waiting for your PR that documents the change you did in https://github.com/kubernetes/kubernetes/pull/57287 in the v1.10 doc :)

Reply:
🤦‍♂️ I wasn't taking "copy" literally I guess. Sounds good, I'll submit a PR as soon as I get a chance!